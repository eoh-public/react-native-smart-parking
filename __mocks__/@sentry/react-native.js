export const setUser = jest.fn();
export const init = jest.fn();

export default {
  init,
  setUser,
};
