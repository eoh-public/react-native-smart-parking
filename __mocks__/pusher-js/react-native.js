const channel = {
  bind: jest.fn(),
};

const pusherClient = {
  subscribe: jest.fn(),
  unsubscribe: jest.fn(),
  disconnect: jest.fn(),
};

pusherClient.subscribe.mockImplementation(() => channel);

const MockPusher = jest.fn();
MockPusher.mockImplementation(() => pusherClient);

export default MockPusher;
