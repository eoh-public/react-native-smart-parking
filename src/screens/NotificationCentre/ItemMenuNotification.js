import React, { memo, useCallback } from 'react';
import { StyleSheet, View, TouchableOpacity } from 'react-native';
import { IconOutline } from '@ant-design/icons-react-native';

import Text from '../../commons/Text';
import { Colors } from '../../configs';
import ViewNotify from '../MapDashboard/components/ViewNotify';

const ItemMenuNotification = memo((props) => {
  const { item, onPressItem } = props;
  const { icon, title, subTitle, number } = item;
  const onPress = useCallback(() => {
    onPressItem && onPressItem(item);
  }, [item, onPressItem]);

  return (
    <TouchableOpacity
      style={styles.container}
      activeOpacity={0.4}
      onPress={onPress}
    >
      {icon}
      <View style={styles.info}>
        <Text type="H4" semibold color={Colors.Gray9}>
          {title}
        </Text>
        <Text type="Label" color={Colors.Gray8} style={styles.subTitle}>
          {subTitle}
        </Text>
      </View>
      <View style={styles.rightComponent}>
        {number && (
          <ViewNotify
            hasNoti
            notiNumber={number}
            styleNotifyNumber={styles.notifiNumber}
          />
        )}
        <IconOutline name="right" size={21} />
      </View>
    </TouchableOpacity>
  );
});

export default ItemMenuNotification;

const styles = StyleSheet.create({
  container: {
    width: '100%',
    paddingHorizontal: 16,
    flexDirection: 'row',
    paddingTop: 16,
  },
  info: {
    flex: 1,
    marginLeft: 26,
    borderBottomWidth: 1,
    borderBottomColor: Colors.Gray4,
    paddingBottom: 16,
  },
  rightComponent: {
    position: 'absolute',
    top: 16,
    right: 16,
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'flex-end',
  },
  subTitle: {
    marginTop: 8,
  },
  notifiNumber: {
    position: 'relative',
    top: 0,
    right: 0,
    width: 25,
    height: 22,
    marginRight: 10,
  },
});
