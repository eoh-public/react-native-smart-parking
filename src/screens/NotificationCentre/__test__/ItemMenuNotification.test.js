import React from 'react';
import { TouchableOpacity } from 'react-native';
import { create, act } from 'react-test-renderer';
import ItemMenuNotification from '../ItemMenuNotification';

describe('Test ItemMenuNotification', () => {
  const item = {
    icon: '',
    title: '',
    subTitle: '',
    number: 1,
  };
  let wrapper;
  test('create ItemMenuNotification', () => {
    act(() => {
      wrapper = create(<ItemMenuNotification item={item} />);
    });
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  test('onPressItem ItemMenuNotification', () => {
    const mockFunc = jest.fn();
    act(() => {
      wrapper = create(
        <ItemMenuNotification item={item} onPressItem={mockFunc} />
      );
    });
    const instance = wrapper.root;
    const button = instance.findByType(TouchableOpacity);
    act(() => {
      button.props.onPress();
    });
    expect(mockFunc.mock.calls.length).toBe(1);
  });
});
