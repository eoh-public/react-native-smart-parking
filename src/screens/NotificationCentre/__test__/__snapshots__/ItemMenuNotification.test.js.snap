// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ItemMenuNotification create ItemMenuNotification 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingTop": 16,
      "width": "100%",
    }
  }
>
  
  <View
    style={
      Object {
        "borderBottomColor": "#E8E8E8",
        "borderBottomWidth": 1,
        "flex": 1,
        "marginLeft": 26,
        "paddingBottom": 16,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#262626",
            "fontFamily": "SFProDisplay-Semibold",
            "fontSize": 16,
            "lineHeight": 24,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#595959",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 12,
            "lineHeight": 20,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "marginTop": 8,
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "position": "absolute",
        "right": 16,
        "top": 16,
      }
    }
  >
    <View
      style={
        Object {
          "overflow": "visible",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F08229",
              "borderColor": "white",
              "borderRadius": 11,
              "borderWidth": 0,
              "height": 22,
              "width": 24,
            },
            undefined,
            Array [
              Object {
                "alignItems": "center",
                "borderWidth": 1,
                "justifyContent": "center",
                "position": "absolute",
                "right": -11,
                "top": -7,
              },
              Object {
                "height": 22,
                "marginRight": 10,
                "position": "relative",
                "right": 0,
                "top": 0,
                "width": 25,
              },
            ],
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 12,
                "lineHeight": 20,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          1
        </Text>
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontFamily": "antoutline",
            "fontSize": 21,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          undefined,
        ]
      }
    >
      î¬º
    </Text>
  </View>
</View>
`;
